app/components/UI/Ramp/Deposit/Views/Modals/RegionSelectorModal/RegionSelectorModal.tsx

```diff
@@ -77,52 +74,24 @@ function RegionSelectorModal() {
       const results = fuseData
         .search(searchString)
         ?.slice(0, MAX_REGION_RESULTS);
-
-      if (results?.length) {
-        return [
-          {
-            title: null,
-            data: results,
-          },
-        ];
-      }
-      return [];
-    }
-
-    const popularRegions = DEPOSIT_REGIONS.filter(
-      (region) => region.recommended,
-    );
-
-    if (popularRegions.length) {
-      return [
-        {
-          title: strings('fiat_on_ramp_aggregator.region.popular_regions'),
-          data: popularRegions,
-        },
-        {
-          title: strings('fiat_on_ramp_aggregator.region.regions'),
-          data: DEPOSIT_REGIONS.filter((region) => !region.recommended),
-        },
-      ];
+      return results || [];
     }
 
-    return [
-      {
-        title: null,
-        data: DEPOSIT_REGIONS,
-      },
-    ];
+    return DEPOSIT_REGIONS.sort((a, b) => {
+      if (a.recommended && !b.recommended) return -1;
+      if (!a.recommended && b.recommended) return 1;
+      return 0;
+    });
```

Comment:
Let's avoid modifying the original reference and return a new array using `toSorted`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted

```suggestion
    return DEPOSIT_REGIONS.toSorted((a, b) => {
      if (a.recommended && !b.recommended) return -1;
      if (!a.recommended && b.recommended) return 1;
      return 0;
    });
```